moralization/transformers_model_manager.py

```diff
@@ -310,12 +318,134 @@ def load_optimizer(self, learning_rate: float = 2e-5, kwargs: dict = None) -> No
+        except AttributeError:
+            raise ValueError(
+                "Dataloader not initialized. Please load the dataloader first."
+            )
+        if not hasattr(self, "optimizer"):
+            raise ValueError(
+                "Optimizer not initialized. Please load the optimizer first."
+            )
+        self.num_training_steps = num_train_epochs * self.num_update_steps_per_epoch
+        self.lr_scheduler = get_scheduler(
+            name=scheduler_name,
+            optimizer=self.optimizer,
+            num_warmup_steps=0,
+            num_training_steps=self.num_training_steps,
+        )
+
+    def postprocess(self, predictions, labels):
+        predictions = predictions.detach().cpu().clone().numpy()
+        labels = labels.detach().cpu().clone().numpy()
+        if not self.label_names:
+            raise ValueError("Label names not set!")
+        # Remove ignored index (special tokens) and convert to labels
+        true_labels = [
+            [self.label_names[m] for m in label if m != -100] for label in labels
+        ]
+        true_predictions = [
+            [self.label_names[p] for (p, m) in zip(prediction, label) if m != -100]
+            for prediction, label in zip(predictions, labels)
+        ]
+        return true_labels, true_predictions
+
+    def train(self) -> None:
+        """Train a model using the pre-loaded components."""
+
+        # show a progress bar
+        progress_bar = tqdm(range(self.num_training_steps))
+
+        for epoch in range(self.num_train_epochs):
+            # set the mode to training
+            self.model.train()
+            for batch in self.train_dataloader:
+                outputs = self.model(**batch)
+                loss = outputs.loss
+                self.accelerator.backward(loss)
+                self.optimizer.step()
+                self.lr_scheduler.step()
+                self.optimizer.zero_grad()
+                progress_bar.update(1)
+
+            # Evaluation
+            self._evaluate_model()
+
+            self.results = self.metric.compute()
+            print(
+                f"epoch {epoch}:",
+                {
+                    key: self.results[f"overall_{key}"]
+                    for key in ["precision", "recall", "f1", "accuracy"]
+                },
+            )
+
+            self.save()
+
+    def _evaluate_model(self):
+        # set mode to evaluation
+        self.model.eval()
+        for batch in self.eval_dataloader:
+            with no_grad():
+                outputs = self.model(**batch)
+            predictions = outputs.logits.argmax(dim=-1)
+            labels = batch["labels"]
+            # Necessary to pad predictions and labels for being gathered
+            predictions = self.accelerator.pad_across_processes(
+                predictions, dim=1, pad_index=-100
+            )
+            labels = self.accelerator.pad_across_processes(
+                labels, dim=1, pad_index=-100
```

Comment:
maybe all these `-100` magic numbers could be replaced with a named variable, e.g. `invalid_word_label`?